src/treemap/iter.rs

```diff
@@ -16,12 +16,40 @@ impl<'a> Iterator for To64Iter<'a> {
     fn next(&mut self) -> Option<u64> {
         self.inner.next().map(|n| util::join(self.hi, n))
     }
+
+    #[inline]
+    fn fold<B, F>(self, init: B, f: F) -> B
+    where
+        Self: Sized,
+        F: FnMut(B, Self::Item) -> B,
+    {
+        #[inline]
+        fn f_with_hi<B>(mut f: impl FnMut(B, u64) -> B, hi: u32) -> impl FnMut(B, u32) -> B {
+            move |b, lo| f(b, ((hi as u64) << 32) + (lo as u64))
+        }
+
+        self.inner.fold(init, f_with_hi(f, self.hi))
```

Comment:
Is the only purpose of this to force inlining the closure? Or is it because you need to pass the `self.hi` as `move`, and then it tries to pass `self` as `move`? Because I would prefer a simpler approach:
```suggestion
        self.inner.fold(init, move |b, lo| f(b, ((self.hi as u64) << 32) + (lo as u64)))
```

Reply:
Maybe we can move this `f_with_hi ` function into the _treemap/util.rs_ module instead of copying/pasting it everywhere and adding a small and helpful description in the process? What do you think?

Reply:
There wasn't much thought given to it from my end. I just did what I saw in adjacent code to `Flatten` in the stdlib.

I've tracked the inlined internal functions down to this PR: https://github.com/rust-lang/rust/pull/62429

The reasoning is making the closures less generic to reduce type length and deduplicate monomorphization. I don't think those apply here, so I'll switch to using the closures directly instead.